src/GraphQL/Extensions/ObjectExtensions.Compiled.cs

```diff
@@ -0,0 +1,456 @@
+using System.Collections;
+using System.Diagnostics;
+using System.Linq.Expressions;
+using System.Reflection;
+using GraphQL.Types;
+
+namespace GraphQL;
+
+public static partial class ObjectExtensions
+{
+    /// <summary>
+    /// Compiles a function to convert a dictionary to an object based on a specified <see cref="IInputObjectGraphType"/> instance.
+    /// The compiled function assumes the passed dictionary object is not <see langword="null"/>.
+    /// </summary>
+    public static Func<IDictionary<string, object?>, object> CompileToObject(
+        [DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicConstructors | DynamicallyAccessedMemberTypes.PublicFields | DynamicallyAccessedMemberTypes.PublicProperties)]
+        Type sourceType,
+        IInputObjectGraphType graphType)
+    {
+        var conv = ValueConverter.GetConversion(typeof(IDictionary<string, object?>), sourceType);
+        if (conv != null)
+            return conv;
+
+        var info = GetReflectionInformation(sourceType, graphType);
+        return CompileToObject(info);
+    }
+
+    /// <summary>
+    /// Compiles a function to convert a dictionary to an object based on a specified <see cref="ReflectionInfo"/> instance.
+    /// </summary>
+    private static Func<IDictionary<string, object?>, object> CompileToObject(ReflectionInfo info)
+    {
+        var bestConstructor = info.Constructor;
+        var ctorFields = info.CtorFields;
+        var members = info.MemberFields;
+
+        // T obj;
```

Comment:
This was the last line I understood in this file. This looks Chinees for me 😅

Reply:
I did try to make it as readable as possible.  Each method has a sample of the code that the 'chinese' compiles to. 😉 One thing that might not be immediately apparent is that the dictionary parameter expression is static and so doesn't get passed to the other methods. (`_dictionaryParam`)

Reply:
i also wrote the code (mostly) in one long multi-line statement, as i think it reads more like the code it compiles to, rather than compiling the innermost expression first, storing it in a variable, and so on.  Then it reads a bit backwards from how it executes.

If you like, I can add comments to each line or so to make it more readable.

Reply:
Everything here is currently `private` so we can rename method names or restructure this code as desired in the future.